rspace/src/main/scala/coop/rchain/rspace/history/Trie.scala

```diff
@@ -2,10 +2,17 @@ package coop.rchain.rspace.history
 
 import coop.rchain.shared.AttemptOps._
 import scodec.Codec
-import scodec.bits.BitVector
+import scodec.bits.{BitVector, ByteVector}
 import scodec.codecs._
 
-sealed trait Trie[+K, +V]                         extends Product with Serializable
+sealed trait Child
```

Comment:
Let's do:

```scala
sealed trait Pointer
case class NodePointer(hash: Blake2b256Hash) extends Pointer
case class LeafPointer(hash: Blake2b256Hash) extends Pointer
case object EmptyPointer extends Pointer
```

I think this naming strategy fits better (we already have `PointerBlock`s).

Reply:
Done